javaslang/src/main/java/javaslang/Value.java

```diff
@@ -660,13 +668,46 @@ default CharSeq toCharSeq() {
      */
     default <K, V> Map<K, V> toMap(Function<? super T, ? extends Tuple2<? extends K, ? extends V>> f) {
         Objects.requireNonNull(f, "f is null");
-        if (isEmpty()) {
-            return HashMap.empty();
-        } else if (isSingleValued()) {
-            return HashMap.of(f.apply(get()));
-        } else {
-            return HashMap.ofEntries(Iterator.ofAll(this).map(f));
-        }
+		return ValueModule.toMap(this, HashMap.empty(), HashMap::of, HashMap::ofEntries, f);
```

Comment:
@danieldietrich I implement such check for `javaslang.ValueModule#toTraversable` because this method convert `Value<T>` into `Traversable<T>`, and in case `Set<T>#toSet()` we can use same object.
But I think this is not needed for `javaslang.ValueModule#toMap`, because we alway have mapping function and in case of converting `HashMap<K, V>` to map we can not return same object because we must consider to mapping function. And only in case than mapping function for new Map return same Tuple we can return original Map. But how to detect this?